### PR TITLE
Mute the robot speaker, microphone and camera

### LINKED ISSUE #29: Handle robot mute (video + audio) from frontend and GUI.
The user should be able to mute the robot directly on the robot's screen. The teleoperator can also control mute at a distance.

### PR DESCRIPTION
**Pull request by joli-1801**
closes #29

### CODE REVIEW

**Review by doumdi — APPROVED** [review]

File: opentera_webrtc_robot_gui/src/ConfigDialog.cpp [hunk 1]
```diff
@@ -1,11 +1,35 @@
...
 ConfigDialog::ConfigDialog(QWidget* parent) : m_ui(new Ui::ConfigDialog())
 {
     m_ui->setupUi(this);
     setWindowFlags(Qt::Dialog | Qt::FramelessWindowHint | Qt::WindowTitleHint);
+
+    //Sliders
+    m_ui->micVolumeSlider->setValue(100);
+    connect(m_ui->micVolumeSlider, &QSlider::valueChanged, (MainWindow*)parent, &MainWindow::_onMicVolumeSliderValueChanged);
+    m_ui->volumeSlider->setValue(100);
+    connect(m_ui->volumeSlider, &QSlider::valueChanged, (MainWindow*)parent, &MainWindow::_onVolumeSliderValueChanged);
```
Comment (philippewarren) [inline opentera_webrtc_robot_gui/src/ConfigDialog.cpp:15]: Don't use C-style casts (Core Guidelines ES.49 https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es49-if-you-must-use-a-cast-use-a-named-cast).
In this case, use either dynamic_cast, or pass a MainWindow* as the constructor argument and then use implicit conversion when you need a QWidget* (suggested in ES.48 https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es48-avoid-casts).

File: opentera_webrtc_demos/scripts/robot_status.py [hunk 2]
```diff
@@ -35,6 +43,15 @@ def get_disk_usage(self, mount_point='/'):
...
 
+    def set_mic_volume(self, msg):
+        self.micVolume = msg.data
+
+    def set_enable_camera(self, msg):
+        self.cameraEnabled = msg.data
+    
+    def set_volume(self, msg):
+        self.volume = msg.data
+
```
Comment (philippewarren) [inline opentera_webrtc_demos/scripts/robot_status.py:54]: Add `_cb` or `_callback` postfix, and maybe a `_` prefix too, to clearly identify that these are internal to the class and not part of its interface

File: opentera_webrtc_demos/scripts/robot_status.py [hunk 3]
```diff
@@ -18,7 +18,15 @@ def __init__(self):
         self.status_webrtc_pub = rospy.Publisher(
             '/webrtc_data_outgoing', String, queue_size=10)
         self.pub_rate = 1
-
+        self.mic_volume_sub = rospy.Subscriber(
+            'mic_volume', Float32, self.set_mic_volume, queue_size=10)
+        self.micVolume = 1
+        self.enable_camera_sub = rospy.Subscriber(
+            'enable_camera', Bool, self.set_enable_camera, queue_size=10)
+        self.cameraEnabled = True
```
Comment (mamaheux) [inline opentera_webrtc_demos/scripts/robot_status.py:26]: Use snake case for micVolume and cameraEnabled.

File: opentera_webrtc_robot_gui/src/ConfigDialog.cpp (hunk 1 above)
Comment (joli-1801) [reply opentera_webrtc_robot_gui/src/ConfigDialog.cpp]: I've changed it to use a dynamic_cast

**Review by philippewarren — APPROVED** [review]

File: opentera_webrtc_robot_gui/src/ConfigDialog.cpp [hunk 4]
```diff
@@ -1,11 +1,47 @@
 #include "ConfigDialog.h"
+#include "MainWindow.h"
 
 #include "ui_ConfigDialog.h"
 
 ConfigDialog::ConfigDialog(QWidget* parent) : m_ui(new Ui::ConfigDialog())
```
Comment (mamaheux) [inline opentera_webrtc_robot_gui/src/ConfigDialog.cpp:6]: Change the type here and remove the dynamic_cast

**Review by mamaheux — APPROVED** [review]

**Pull request merged**